### PR TITLE
Improve Loki logs render with ANSI colors

### PR DESCRIPTION
**Pull request by avaly**
This is a follow-up, based on the comments from https://github.com/grafana/grafana/pull/15305

When using Loki, we now detect whether a log entry has ANSI codes during the merging of data streams in the Redux reducer.

Also, during render, the search and parse highlighting get priority over the ANSI colors rendering.

Unrelated: fixed a spelling typo in `LogLevel.unkown`

With ANSI codes, no search:

selection_2019-02-21_754x23

With ANSI codes, but with search:

selection_2019-02-21_754x23 001

Unfortunately, due to how the `react-hightlight-words` component is working internally, we can't combine the search words with the ANSI colors, so we have to use the raw log entry when rendering the search component. A future workaround for this issue would be to create a highlighter component that understands both cases: search words and ANSI color codes.

### CODE REVIEW

**Comment by avaly** [comment]
cc @davkal and @torkelo

File: public/app/plugins/datasource/loki/result_transformer.ts [hunk 1]
```diff
@@ -133,7 +136,9 @@ export function processEntry(
     timeEpochMs,
     timeLocal,
     uniqueLabels,
+    hasAnsi,
     entry: line,
+    raw: hasAnsi ? ansicolor.strip(line) : line,
```
Comment (davkal) [inline public/app/plugins/datasource/loki/result_transformer.ts]: If we introduce a `raw` property, it makes more sense for it have the `line` value, and then use the stripped version in the `entry` property.

File: public/app/plugins/datasource/graphite/FunctionEditor.tsx [hunk 2]
```diff
@@ -23,7 +23,13 @@ class FunctionEditor extends React.PureComponent<FunctionEditorProps, FunctionEd
   }
 
   renderContent = ({ updatePopperPosition }) => {
-    const { onMoveLeft, onMoveRight, func: { def: { name, description } } } = this.props;
+    const {
```
Comment (avaly) [inline public/app/plugins/datasource/graphite/FunctionEditor.tsx]: This change was made due to `prettier:check` failing on this file.

**Review by davkal — APPROVED** [review]
Great, thanks for fixing that!

**Pull request merged**